src/app/components/connect-addon-modal/connect-addon-modal.coffee

```diff
@@ -0,0 +1,91 @@
+angular.module('mnoEnterpriseAngular').component('connectAddonModal', {
+  bindings: {
+    resolve: '<'
+    close: '&',
+    dismiss: '&'
+  },
+  templateUrl: 'app/components/connect-addon-modal/connect-addon-modal.html',
+  controller: (MnoeAppInstances, toastr, $translate) ->
+
+    ctrl = this
+
+    ctrl.app = null
+    ctrl.model = {}
+    ctrl.hasLinked = false
+    ctrl.hasChosenEntities = false
+    ctrl.historicalData = false
+    ctrl.isFormLoading = true
+    ctrl.isSubmitting = false
+    ctrl.historicalData = false
+    ctrl.date = new Date()
+
+    $translate([
+      'mno_enterprise.templates.components.addon_connect.sync.sync_launched',
+      'mno_enterprise.templates.components.addon_connect.link_account.submit_form',
+      'mno_enterprise.templates.components.addon_connect.entities.update_entities',
+      'mno_enterprise.templates.components.addon_connect.sync.start_sync',
+    ]).then((tls) ->
+      ctrl.syncLaunchedTl = tls['mno_enterprise.templates.components.addon_connect.sync.sync_launched']
+      ctrl.submitTl = tls['mno_enterprise.templates.components.addon_connect.link_account.submit_form']
+      ctrl.updateTl = tls['mno_enterprise.templates.components.addon_connect.entities.update_entities']
+      ctrl.startSyncTl = tls['mno_enterprise.templates.components.addon_connect.sync.start_sync']
+    )
+
+    ctrl.$onInit = ->
+      ctrl.app = ctrl.resolve.app
+      ctrl.hasLinked = ctrl.app.addon_organization.has_account_linked
+      MnoeAppInstances.getForm(ctrl.app)
+        .then((response) ->
+          ctrl.schema = response.schema
+          ctrl.form = [ "*" ]
+          ctrl.isFormLoading = false
+        )
+
+    ctrl.close = ->
+      ctrl.close()
+
+    ctrl.submit = (form) ->
+      ctrl.isSubmitting = true
+      MnoeAppInstances.submitForm(ctrl.app, ctrl.model)
+        .then((response) ->
+          if response.error
+            toastr.error(response.error)
```

Comment:
Shouldn't the error be catched in an `error` callback?

Reply:
We have some validation on the connector side when a user wants to link their account.
For example, only one account can be linked, or sometimes, we know of some kind of common mistake people make. And so, for each of these situations, we have a custom error message that I returned in the response.
Should I change the json key from the response "error" to something like "linking_error"?

Reply:
Yes, that would be great. This is a bit confusing and alarming. `linking_error` sounds more like a specific error.